core/src/main/java/org/infinispan/Cache.java

```diff
@@ -323,4 +322,15 @@
     */
    @Override
    CloseableIteratorSet<Entry<K, V>> entrySet();
+
+   /**
+    * Removes all mappings from the cache.
+    * <p/>
+    * <b>Note:</b> It assumes no other operation running in the cache during its invocation. If not the case, data can
```

Comment:
IMO this is pretty much the most important changes in this PR.  I wonder if we want to emphasize this more by saying something like "This should never be invoked in production unless you can guarantee no other invocations are ran concurrently."